### PR TITLE
style: Fix self-assigning-variable (PLW0127)

### PR DESCRIPTION
**Pull request by echoix**
Ruff rule: https://docs.astral.sh/ruff/rules/self-assigning-variable/

12 instance fixed, no auto fixes were available.
I think I found a bug that was apparent from the syntax highlighting and context. I added a note, but could be removed from this PR after some input.

In some files from temporal, I think by context that the self-assigned variable was meant to be the map id as in the "else" branch. Otherwise, the variable with `_id` wouldn't be available at all if the name contained "@" when doing for example `upper_strds = tgis.SpaceTimeRasterDataset(upper_id)`.

### CODE REVIEW

**Comment by echoix** [comment]
There's also a collapsible else if that will be suggested, but I want feedback to what line to apply the lost comment

File: temporal/t.rast.accumulate/t.rast.accumulate.py [hunk 1]
```diff
@@ -244,6 +244,9 @@ def main():
 
     # The lower threshold space time raster dataset
     if lower:
+        # TODO: Check if range is defined elsewhere than this file.
+        # Otherwise, the code is unreachable as range is a class, built into Python,
+        # used in for loops notably. See t.rast.accdetect for usage of range_
```
Comment (echoix) [inline temporal/t.rast.accumulate/t.rast.accumulate.py:249]: image
image

We also can see from my fork's PR for code coverage also show that this part of code isn't tested:
https://app.codecov.io/gh/echoix/grass/pull/168/blob/temporal/t.rast.accumulate/t.rast.accumulate.py
image

**Review by echoix — COMMENTED** [review]
I added some more info on the possible issue, where a variable named "range" isn't defined anywhere, but used, and doesn't fail because it matches the built in "range(start, stop)" class https://docs.python.org/3/library/stdtypes.html#range

File: temporal/t.rast.accumulate/t.rast.accumulate.py (hunk 1 above)
Comment (ninsbl) [reply temporal/t.rast.accumulate/t.rast.accumulate.py:249]: After looking into `t.rast.accdetect` and comparing it to the code here, my conclusion is that the codeblock behind `if not range:` is a copy-paste leftover and can be removed. As you say it is never reached and here the value range is defined by upper and lower...

Comment (echoix) [reply temporal/t.rast.accumulate/t.rast.accumulate.py:249]: So I think it might be best to remove it in another PR

Comment (echoix) [reply temporal/t.rast.accumulate/t.rast.accumulate.py:249]: See #4154

Comment (echoix) [reply temporal/t.rast.accumulate/t.rast.accumulate.py:249]:
```suggestion
```

**Comment by echoix** [comment]
#4154 was merged, and this PR reverted the part where I added a TODO (to have a discussion). So the PR is ready to review+merge

**Review by wenzeslaus — APPROVED** [review]

**Pull request merged**